### PR TITLE
fix: tty notification and verbose hooks

### PR DESCRIPTION
**Pull request by ichoosetoaccept**
## Summary

Fix template update notification being swallowed by `gt` during `post-checkout` hook execution, add verbose output to pytest hooks, and add comprehensive test coverage.

## Problem

The `check-template-update.sh` script writes notifications to stdout, which `gt` (Graphite CLI) captures and discards when wrapping `git checkout`. Even with `verbose = true` in prek, `gt` sits above prek and swallows everything. `prek` itself delivers the output correctly with `verbose = true`.

### Script fix — `/dev/tty` output (`project/scripts/check-template-update.sh`)
- Probe `/dev/tty` writability with a subshell test: `(echo -n > /dev/tty) 2>/dev/null`
- Write notification directly to `/dev/tty` when available (bypasses all pipe capture)
- Fall back to stdout for non-interactive contexts (CI, scripts, tests)

### Verbose pytest hooks (`prek.toml`, `project/prek.toml.jinja`)
- Add `verbose = true` to `pytest-testmon` and `pytest-cov` hooks in generated projects
- Add `verbose = true` to `pytest-cov` hook in template project
- Change `-q` to `-v` in template project's pytest-cov entry so test names are visible

### Test coverage (`tests/test_template.py`)
- **12 fast tests** (< 2s total) in `TestTemplateUpdateNotification` using mock curl and a minimal directory (script is static, no jinja — no need for full `copytree`)
- **1 slow integration test** verifying raw git hook delivers output via `/dev/tty` fallback
- Covers: output channels, hook arguments, cooldown mechanism, manual invocation bypass, edge cases (empty commit, API failure, `.git` suffix), and git hook integration
- Fix cache hash to match shell's `echo "$PWD" | sha1sum` (trailing newline + `/tmp`)
- Add `start_new_session=True` to all subprocess calls invoking the script to prevent `/dev/tty` leak during test runs

## Testing

```
$ poe test  # 11 fast notification tests included
$ poe test-all  # + 1 slow git integration test
```

132 tests pass, 96% coverage.

### CODE REVIEW

File: tests/test_template.py [hunk 1]
```diff
@@ -695,11 +728,241 @@ def test_script_silent_when_up_to_date(self, copier_defaults: dict, project_fact
...
+            ["git", "checkout", "main"],
+            cwd=project,
+            capture_output=True,
+            text=True,
+            check=False,
+            env=git_env,
+            start_new_session=True,
+        )
+        combined = result.stdout + result.stderr
+        assert "Template update available" in combined
```
[the remark this answers is not in the document]
Comment (ichoosetoaccept) [reply tests/test_template.py]: Good catch on the fragility. `COPIER_CHECK_INTERVAL=0` is intentional and documented in the hook template — it's the "always check" override. A minimum interval floor would be a separate design decision. Acceptable for now.
